README.md

```diff
@@ -21,135 +21,169 @@ gem 'acme-client'
 ```
 
 ## Usage
+* [Setting up a client](#setting-up-a-client)
+* [Account management](#account-management)
+* [Obtaining a certificate](#obtaining-a-certificate)
+  * [Ordering a certificate](#ordering-a-certificate)
+  * [Completing an HTTP challenge](#preparing-for-http-challenge)
+  * [Completing an DNS challenge](#preparing-for-dns-challenge)
+  * [Request a challenge verification](#request-a-challenge-verification)
+  * [Downloading a certificate](#downloading-a-certificate)
+* [Extra](#extra)
+  * [Certificate revokation](#certificate-revokation)
+  * [Certificate renewal](#certificate-renewal)
 
-### Register client
+## Setting up a client
 
-In order to authenticate our client, we have to create an account for it.
+The client is initialized with a private key and the directory of your ACME provider.
+
+LetsEncrypt's `directory` is `https://acme-v01.api.letsencrypt.org/directory`
+
+`acme-ruby` expects `OpenSSL::PKey::RSA` or `OpenSSL::PKey::EC`
+
+You can generate one in Ruby using OpenSSL.
 
 ```ruby
-# We're going to need a private key.
 require 'openssl'
 private_key = OpenSSL::PKey::RSA.new(4096)
+```
 
-# We need an ACME server to talk to, see github.com/letsencrypt/boulder
-# WARNING: This endpoint is the production endpoint, which is rate limited and will produce valid certificates.
-# You should probably use the staging endpoint for all your experimentation:
-# endpoint = 'https://acme-staging.api.letsencrypt.org/'
-endpoint = 'https://acme-v01.api.letsencrypt.org/'
+Or load one from a PEM file
 
-# Initialize the client
-require 'acme-client'
-client = Acme::Client.new(private_key: private_key, endpoint: endpoint, connection_options: { request: { open_timeout: 5, timeout: 5 } })
+```ruby
+require 'openssl'
+OpenSSL::PKey::RSA.new(File.read('/path/to/private_key.pem'))
+```
+
+See [RSA](https://ruby.github.io/openssl/OpenSSL/PKey/RSA.html) and [EC](https://ruby.github.io/openssl/OpenSSL/PKey/EC.html) for documentation.
+
+
+```ruby
+client = Acme::Client.new(private_key: private_key, directory: 'https://acme-v01.api.letsencrypt.org/directory')
+```
 
-# If the private key is not known to the server, we need to register it for the first time.
-registration = client.register(contact: 'mailto:contact@example.com')
+If your account is already registered, you can save some API calls by passing your key ID directly. This will avoid an unnecessary API call to retrieve it from your private key.
 
-# You may need to agree to the terms of service (that's up the to the server to require it or not but boulder does by default)
-registration.agree_terms
+```ruby
+client = Acme::Client.new(private_key: private_key, directory: 'https://acme-v01.api.letsencrypt.org/directory', kid: 'https://example.com/acme/acct/1')
```

Comment:
Question for the sake of my understanding: I read the ACME v2 specs, and I wonder... is `kid` simply a custom string identifier?

Reply:
What about an existing account? Can I fetch it in some ways?

Reply:
@weppos From a protocol standpoint there is a way. You can use the mechanism described in [Section 7.3.1](https://tools.ietf.org/html/draft-ietf-acme-acme-10#section-7.3.1) and POST the `newAccount` endpoint sending a signed message body `{"onlyReturnExisting": true}`. The JWS that authenticates the request should embed the JWK of the account key you want to learn the key ID for. If the key is known on the server side you'll get back the KeyID.

From a `unixcharles/acme-client` perspective I don't know, sorry :-) @unixcharles probably does!

Reply:
> From a unixcharles/acme-client perspective I don't know, sorry :-) @unixcharles probably does!

If the `kid` is not provided at initialization, the client will lazy load the `kid` by doing a `POST` to `newAccount` whenever the `kid` is required.

So you can just instantiate  a new client with the private key without the `kid` and call `Acme::Client#kid` on it to retrieve it.

Good input, I'll add instruction for this.